packages/service-worker/worker/src/driver.ts

```diff
@@ -454,7 +454,14 @@ export class Driver implements Debuggable, UpdateSource {
 
     const onActionClick = notification?.data?.onActionClick[notificationAction];
 
-    const urlToOpen = new URL(onActionClick?.url ?? '', this.scope.registration.scope).href;
+    let urlToOpen = new URL(onActionClick?.url ?? '', this.scope.registration.scope).href;
+
+    const namespaceConfig = this.getNamespaceConfig();
+
+    if (this.isNamespaceSupported(namespaceConfig, notification?.data?.url)) {
+      let url = new URL(namespaceConfig!.url, this.scope.registration.scope).href;
+      urlToOpen = `${url}?${namespaceConfig!.param}=${encodeURIComponent(notification.data.url)}`;
```

Comment:
Aquí me parece un poco peligroso añadirle la '?' ya que no sabemos como es la `namespaceConfig.url`

```suggestion
      let url = new URL(namespaceConfig!.url, this.scope.registration.scope);
      url.queryparams.append(namespaceConfig!.param, encodeURIComponent(notification.data.url));
      urlToOpen = url.href;
```